### PR TITLE
more modular /customizable script. Full refactor.

### PR DESCRIPTION
**Pull request by holylander**
This PR that is kind of big contains:
- update on doc
- script starter, in which user may predefine the required parameters
- a (mostly) full refactor of the script, so it is more compatible with linux, and more customizable, in terms of:
  -  output of the logs
  - target folder
  - verbose messages

### CODE REVIEW

File: extract-git-logs/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# ignore local version of running script 
+run-extract-logs.local.ps1
+
+# ignore temp folder
+temp/**/*
```
Comment (philzartis) [inline extract-git-logs/.gitignore:5]: temp === temp/**/* ;)

File: extract-git-logs/run-extract-logs.template.ps1 [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+$fnParams = @{
+    gitAuthor         = "" # use when you want to start with a specific user ( app will promt u otherwise )
+    targetBranch      = "master"   # look for logs on this branch
+    semester          = -1  # available values: 0 (full year ) 1 ( 1st semester ) 2 ( 2nd semester ). ( app will promt u otherwise )
+    targetYear        = "" # define target year( app will promt u otherwise )
+    nestedFolders     = 2 # define how many nested folder level the script should analysis ( app will use "4" as default )
+    allProjectsFolder = "" # put the full path of the folder where your working git repos live in
+    # for instace: "C:\Users\[user]\Documents\vscode"
+    # or: "/home/[user]/vscode"
+    debuginfo         = $false # wheter to print or not extra info
```
Comment (philzartis) [inline extract-git-logs/run-extract-logs.template.ps1:13]: comment optional config values

**Review by philzartis — APPROVED** [review]

**Comment by jcarrey** [comment]
@holylander Would you mind continuing with this PR ?

**Comment by holylander** [comment]
@jcarrey done ;)

**Pull request merged**